### PR TITLE
scbpf: add basic eBPF map handling tool

### PR DESCRIPTION
**Pull request by regit**
SCbpf part of #3716 

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Describe changes:
- basic tool to handle IPv4 maps

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/440
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/221

### CODE REVIEW

File: ebpf/scbpf/scbpf.c [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#include <linux/unistd.h>
```
Comment (victorjulien) [inline ebpf/scbpf/scbpf.c]: Should this have a copyright header? Also, coding style is different from Suricata's.

Comment (regit) [reply ebpf/scbpf/scbpf.c]: Indeed.

**Comment by regit** [comment]
@victorjulien are you ok with the concept of adding this tool inside Suricata source code? Alternatively we could have it as a separate tool to allow different release cycle.

**Comment by victorjulien** [comment]
I guess my first thought was: can this be part of suricatactl? But I'm not sure if it feasible to convert this to python?

**Comment by regit** [comment]
It would not be really trivial. It may be doable. I need to look at it to see if we can get something done without pulling half of the Python planet in term of dependencies.

**Comment by jasonish** [comment]
Would it make sense to have it as a sub-command of Suricata? ./suricata
--scbpf or something like that?

**Comment by regit** [comment]
@jasonish we need all insert/update/delete/list operations, looks like a lot of option to add to Suricata main code. Not fan of it even more because I plan some contributions from beginner on it (intern). Python could be a nice choice, I'm gonna check if we can use that.

**Comment by jasonish** [comment]
> @jasonish we need all insert/update/delete/list operations, looks like a lot of option to add to Suricata main code. Not fan of it even more because I plan some contributions from beginner on it (intern). Python could be a nice choice, I'm gonna check if we can use that.

Makes sense.  So with Python stuff (suricatactl, suricata-update) I've been very sensitive to dependencies that are required. And we've started to add some checks to configure.ac.  But I don't think we have to add all dependencies to configure.ac, just the core stuff.

So if you had this in Python, "suricatactl scebpf" or whatever, in my opinion I think it would be fine to error out with an error "such and such python module required".  I'd be curious to see what Victor thinks, but I don't see this as a "core" feature that would should check all deps for during install of Suricata.

**Comment by victorjulien** [comment]
Is this still relevant now that ```bpfctrl``` exists?

**Comment by regit** [comment]
Can be closed as we can use bpfctrl.

**Pull request closed without merging**